### PR TITLE
Add coveralls badge

### PR DESCRIPTION
**Pull request by danielo515**
This adds the coveralls badge.
IT is built on top of the previous PR #22 

- Added a coverage task. This is required by Travis CI, and it is run after everything suceeds
- Added coverall as a dev dependency
- Added coverall as npm script 
- Added coverall execution to travis file

To see a working module generated using the proposed changes, please check:
https://github.com/danielo515/generator-module-boilerplate-test-es6/tree/add-coverall-badge

### CODE REVIEW

**Comment by thibmaek** [comment]
Think it might be a bit opinionated adding coveralls directly into a PR without first considering possible options. If I recall it correctly some issues on this repo list possible options for a coverage service. Might be worth waiting on @duivvv response (that is after everyone enjoyed Christmas dinner of course 😉 ) on which code coverage system to pick

**Comment by danielo515** [comment]
Sorry @thibmaek , I just added the one that fit my needs and made a pull request. I didn't think much about it. My bad. It was not my intention to be rude

**Comment by duivvv** [comment]
no problem, will look into it a bit later :)

**Comment by duivvv** [comment]
I do like coveralls, but in the confusion of their login I assumed you need the 5$ subscription.
But looks like open source repo's are completely free.

Will probably merge this :)

**Comment by danielo515** [comment]
>  in the confusion of their login I assumed I needed the 5$ subscription (even for open source projects)

Happened the same to me, but open source projects, as you said, are free.

Regards

**Comment by duivvv** [comment]
restructured / changed some internals, can you resolve the conflicts
I will merge it

**Comment by duivvv** [comment]
@danielo515

**Comment by danielo515** [comment]
Sorry, I missed the notification. Will try to resolve them tonight

**Comment by danielo515** [comment]
Let me know if conflicts went away...

**Comment by danielo515** [comment]
I'm curious, why have you added loose true to the babel config of the tests? There is no need to make them backwards compatible, is just for running.

File: generators/app/templates/_package.json [hunk 1]
```diff
@@ -31,6 +31,9 @@
     "test:coverage": "jest --coverage ",
     "test:watch": "clear && jest --watch",
 
+    "coveralls": "cat ./coverage/lcov.info | ./node_modules/.bin/coveralls",
+    "coverage": "jest --coverage ",
```
Comment (duivvv) [inline generators/app/templates/_package.json]: we already have test:coverage, no need to define it 2 times :)

File: generators/app/templates/README.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 [![NPM](https://img.shields.io/npm/v/<%= name %>.svg?style=flat-square)](https://www.npmjs.com/package/<%= name %>)
 [![Travis](https://img.shields.io/travis/<%= github %>/<%= name %>/master.svg?style=flat-square)](https://travis-ci.org/<%= github %>/<%= name %>)
 [![David](https://img.shields.io/david/<%= github %>/<%= name %>.svg?style=flat-square)](https://david-dm.org/<%= github %>/<%= name %>)
+[![Coverage Status](https://coveralls.io/repos/<%= github %>/<%= name %>/badge.svg?branch=master)](https://coveralls.io/r/<%= github %>/<%= name %>?branch=master)
```
Comment (duivvv) [inline generators/app/templates/README.md]: could you use the shields.io one (with flat style) for consistency

Comment (danielo515) [inline generators/app/templates/README.md]: OK, will try. Never realized

File: generators/app/templates/_package.json (hunk 1 above)
Comment (danielo515) [inline generators/app/templates/_package.json]: This is automatically run by Travis C I and it is required to be named coverage. Maybe we could just call `test:coverage` from within?

Comment (danielo515) [inline generators/app/templates/_package.json]: It is used to report the coverage to coverall, missed to mention that fact.

Comment (duivvv) [inline generators/app/templates/_package.json]: travis runs `npm run lint:test` after that the coverage folder exists and npm run coveralls upload the results to coveralls

works fine, tested it with https://github.com/duivvv/note-to-frequency

**Comment by danielo515** [comment]
Dear @duivvv , thank you for pointing my problems out. I just changed the PR according to your suggestions

**Comment by duivvv** [comment]
Thanks :tada:

**Pull request merged**